lib/rules/no-trailing-spaces.js

```diff
@@ -25,25 +27,31 @@ module.exports = function(context) {
 
         "Program": function checkTrailingSpaces(node) {
 
-            // Let's hack. Since Esprima does not return whitespace nodes,
-            // fetch the source code and do black magic via regexps.
+            // Let's hack. Since Espree does not return whitespace nodes,
+            // fetch the source code and do matching via regexps.
 
             var src = context.getSource(),
-                re = new RegExp((skipBlankLines ? BLANK_PREFIX : "") + TRAILER, "mg"),
-                match, lines, location;
+                re = new RegExp(NONBLANK),
+                skipMatch = new RegExp(SKIP_BLANK),
+                matches, lines = src.split(/\r?\n/), location;
 
-            while ((match = re.exec(src)) !== null) {
-                lines = src.slice(0, re.lastIndex).split(/\r?\n/g);
+            for (var i = 0, ii = lines.length; i < ii; i++) {
+                if (skipBlankLines && skipMatch.test(lines[i])) {
+                    continue;
+                }
+                matches = re.exec(lines[i]);
```

Comment:
I think I would swap the order of those tests. First check if it has any trailing spaces, then check if it's a blank line, only if it has trailing spaces. It's a good assumption that number of lines with content is going to be greater then number of empty lines, and number of lines with trailing spaces will be less then full number of lines. So in the end two RegEx will execute less often (only for lines with trailing spaces or empty lines).